### PR TITLE
fixing corner cases of config refactoring

### PR DESCRIPTION
**Pull request by alexander-fenster**
This PR fixes two corner cases that were causing smoke tests failures (mostly with `artman_base.yaml`) after recent config refactoring. The two corner cases that this PR fixes are:
- empty `proto_deps`
- no `service_yaml`

### CODE REVIEW

File: artman/tasks/gapic_tasks.py [hunk 1]
```diff
@@ -34,11 +34,13 @@ def execute(self, toolkit_path, descriptor_set, service_yaml,
         self.exec_command(['mkdir', '-p', config_gen_dir])
         config_gen_path = os.path.join(config_gen_dir,
                                        api_full_name + '_gapic.yaml')
-        service_args = ['--service_yaml=' + os.path.abspath(service_yaml)]
         args = [
             '--descriptor_set=' + os.path.abspath(descriptor_set),
             '--output=' + os.path.abspath(config_gen_path)
-        ] + service_args
+        ]
+        if service_yaml != '':
```
Comment (garrettjonesgoogle) [inline artman/tasks/gapic_tasks.py]: If I understand correctly, empty strings in Python are falsy, so you could just write:

`if service_yaml`:

File: artman/tasks/gapic_tasks.py [hunk 2]
```diff
@@ -34,11 +34,13 @@ def execute(self, toolkit_path, descriptor_set, service_yaml,
...
+          service_args = ['--service_yaml=' + os.path.abspath(service_yaml)]
+          args = args + service_args
```
Comment (garrettjonesgoogle) [inline artman/tasks/gapic_tasks.py]: How about:

`args += ['--service_yaml=' + os.path.abspath(service_yaml)]`

File: artman/tasks/gapic_tasks.py [hunk 3]
```diff
@@ -34,11 +34,12 @@ def execute(self, toolkit_path, descriptor_set, service_yaml,
...
         config_gen_path = os.path.join(config_gen_dir,
                                        api_full_name + '_gapic.yaml')
-        service_args = ['--service_yaml=' + os.path.abspath(service_yaml)]
         args = [
             '--descriptor_set=' + os.path.abspath(descriptor_set),
             '--output=' + os.path.abspath(config_gen_path)
-        ] + service_args
+        ]
+        if service_yaml:
+          args = args + ['--service_yaml=' + os.path.abspath(service_yaml)]
```
Comment (garrettjonesgoogle) [inline artman/tasks/gapic_tasks.py]: Indentation needs to be 4 spaces

Comment (alexander-fenster) [reply artman/tasks/gapic_tasks.py]: oh. something strange happened
